src/js/core.js

```diff
@@ -923,14 +923,38 @@ function MediumEditor(elements, options) {
                 i;
 
             if (opts.url && opts.url.trim().length > 0) {
-                this.options.ownerDocument.execCommand('createLink', false, opts.url);
-
-                if (this.options.targetBlank || opts.target === '_blank') {
-                    Util.setTargetBlank(Selection.getSelectionStart(this.options.ownerDocument), opts.url);
-                }
+                var currentSelection = this.options.contentWindow.getSelection();
+                if (currentSelection) {
+                    var exportedSelection = this.exportSelection(),
+                        currentEditor,
+                        startContainerParentElement,
+                        endContainerParentElement,
+                        textNodes;
+
+                    currentEditor = Util.traverseUp(currentSelection.getRangeAt(0).startContainer, function (node) {
+                        return node.getAttribute('data-medium-editor-element');
+                    });
+                    startContainerParentElement = Util.findParentBlockContainer(
+                        currentSelection.getRangeAt(0).startContainer);
+                    endContainerParentElement = Util.findParentBlockContainer(
+                        currentSelection.getRangeAt(0).endContainer);
+
+                    if (startContainerParentElement === endContainerParentElement) {
```

Comment:
The goal of this code is to target the most common case (selection within 1 paragraph) and fix the issues that occur when the selection partially overlaps common inline elements such as SPAN, STRONG, and EM.

It isn't possible to fix these issues if the selection spans multiple parent (block) elements as HTML doesn't allow a single anchor tag to begin within one block element and continue into partial content of a subsequent block element, there has to be individual anchor tags inside each block element. E.g. `<p>Hello <a href="foo">world,</p><p>This is a test.</a> But this isn't</p>` is not allowed in HTML. So in that case we just use the browser's built-in `createLink` as there isn't much special we can do to fix stuff.